### ISSUE TITLE
Use with aws-vault in version 4.X.X is broken

### ISSUE THREAD

**Issue by npapapietro** [lensapp/lens#2115]
**Describe the bug**
Use with aws-vault in version 4.X.X is broken

**To Reproduce**
1. Install version 3.6.9 run `aws-vault exec <my-profile> -- lens --context <associated kubecontext>`
2. Able to login and use specific cluster associated to my aws-vault profile and kube context
3. Upgrade to latest lens and repeat aws-vault command.
4. See error about "please run aws configure". Lens is not picking up credentials from aws-vault

**Environment (please complete the following information):**
- Lens Version: 4.X.X
- OS: Windows
- Installation method (e.g. snap or AppImage in Linux): Choco

**Additional context**
The same command runs across the board for kubernetes apps k9s, kubectl, fluxctl, lens-v3.6.9

**Comment by jim-docker** [comment]
Did you try removing the cluster from Lens and adding it back?

**Comment by npapapietro** [comment]
So this was just setting up lens to begin with after updating. I ran `aws eks update-kubeconfig` to get the latest (all aws cli commands are ran using `aws-vault`) and then lens was able to see the cluster selection as an option to add a new cluster. The result was unauthorized.

I tested by hardcoding my credentials into `~/.aws/credentials` , which allowed connection.  I then repeated my `aws-vault` steps on a different machine (I was setting it up for a coworker). I had the same result. We both rolled back to v3.X.X and it fixed the issue.

**Comment by chenhunghan** [comment]
Hi, @npapapietro

I guess you saw this screen in Lens?

<img width="770" alt="Screen Shot 2021-02-23 at 1 18 35 PM" src="https://user-images.githubusercontent.com/1474479/108836607-b6ca5b00-75d9-11eb-9cbc-5275c54c29bc.png">

I had similar issue lately, I think the issue is from `aws-vault`, not from Lens. Please check your `kubeconfig` (generated by `aws-vault`, normally at `~/.kube/config`)

I notice that if the aws cli in used is "wrapped" with `aws-vault`, the `kubeconfig` generated will missing `AWS_PROFILE`, for example,

Using `aws-cli` with `aws-vault`

```bash
aws-vault exec <my-profile> -- aws eks --region eu-west-1 update-kubeconfig --name my-cluster
```

the kubeconfig `aws-cli` generated will be something like

```yaml
apiVersion: v1
clusters:
- cluster:
     certificate-authority-data: ...
     name: my-cluster
kind: Config
users:
- name: aws-account/my-cluster
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - eu-west-1
      - eks
      - get-token
      - --cluster-name
      - my-cluster
      command: aws
```

but if without `aws-vault`

```bash
aws eks --profile <my-profile> --region eu-west-1 update-kubeconfig --name my-cluster
```

similar kubesonfig is updated, but this time with `AWS_PROFILE` as `env`

```yaml
apiVersion: v1
clusters:
- cluster:
     certificate-authority-data: ...
     name: my-cluster
kind: Config
users:
- name: aws-account/my-cluster
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - eu-west-1
      - eks
      - get-token
      - --cluster-name
      - my-cluster
      command: aws
      env:
      - name: AWS_PROFILE
        value: my-aws-profile
```

TL;DR:

please check your kubeconfig, to see if 

```yaml
env:
      - name: AWS_PROFILE
        value: my-aws-profile
```

is missing.

**Comment by npapapietro** [comment]
@chenhunghan You are right, my kubeconfig looks like 
```yaml
- name: <arn of cluster>
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - us-east-2
      - eks
      - get-token
      - --cluster-name
      - <cluster name>
      command: aws2
      env: null
```
I suspected something changed with lens since the issue only starts in 4.X, but it might be aws-vault wasn't correctly generated the config and Lens <=3.X was never checking that. I'll go over there and see if I can get some answers

**Comment by chenhunghan** [comment]
from my (limited) understanding of Lens, it won't edit your kubeconfig file (but aws-vault may), maybe it was working because you have set AWS_PROFILE in your shell.

**Comment by Nokel81** [comment]
Is this still an issue? Lens never modifies your Kubeconfig.

**Comment by npapapietro** [comment]
I am still having a problem, my work around was to just hardcode my credentials in `~/.aws/config` which is not really convenient with multiple cloud accounts. 

It doesn't seem to be the fact that lens modifies the kubconfig. When you run the update-kubeconfig with aws-vault, there is not aws profile set because everything is read in from the environment varibales.

I could be wrong but I believe this is just order of checking where AWS credentials currently are.

**Comment by npapapietro** [comment]
So I was able to fix it by running
```bash
aws-vault exec --no-session <profile> -- cmd
lens --context <kube context>
```

**Comment by os6sense** [comment]
Lots of different variants along @npapapietro's lines e.g   `aws-vault exec <profile> -- kontena-lens --context=<context>` works for me on Ubuntu 20.04, Lens 4.2.0, aws-vault 5.4.4/6.3.1. I already had aws-vault/kubectl setup and working with MFA. The biggest confusion for me was the naming of the lens install.

**Comment by Nokel81** [comment]
Yeah... that has to do with the fact that Lens used to be owned by a company called Kontena, I don't think snap likes when you change the executable's name so we haven't been able to remove it completely from the code base.
